Plugins/NativeCamera/Source/NativeCamera.h

```diff
@@ -0,0 +1,30 @@
+#pragma once
+#include <bgfx/bgfx.h>
+
+namespace Babylon::Plugins::Internal
+{
+    // Move this struct to Graphics
+    struct TextureData final
+    {
+        ~TextureData()
+        {
+            if (bgfx::isValid(Handle))
+            {
+                bgfx::destroy(Handle);
+            }
+        }
+
+        bgfx::TextureHandle Handle{ bgfx::kInvalidHandle };
+        uint32_t Width{ 0 };
+        uint32_t Height{ 0 };
+        uint32_t Flags{ 0 };
+        uint8_t AnisotropicLevel{ 0 };
+    };
+
+    struct CameraInterface
+    {
+        virtual ~CameraInterface() = default;
+        static CameraInterface* CreateInterface(uint32_t width, uint32_t height, bool frontCamera);
+        virtual void UpdateCameraTexture(bgfx::TextureHandle textureHandle) = 0;
```

Comment:
Are the virtuality, staticness, and raw pointer necessary? Unless I'm misunderstanding, there's never more than one implementation of this at a time (one per platform). If so, could you just declare an ordinary constructor and methods, then have platform-specific implementations of them without virtuality/staticness/raw pointers?

```
CameraInterface(uint32_t, uint32_t, bool);
~CameraInterface();
void UpdateCameraTexture(bgfx::TextureHandle);
```

Reply:
That was my first guess but the allocation also contains specific data per platform. It's not like the shader compiler where only code per platform is different and data the same.

Reply:
As an alternative, you could probably do that with a pImpl-like pattern, I think. Forward declare `Impl`, or `CameraInterfaceState`, or something like that in the header, give `CameraInterface` a unique pointer to it, then define the struct only in the platform-specific cpp file. I haven't written it out, but I think that might give you the best of both worlds.

Reply:
If I follow you, I create a new CameraInterface structure which will contain a pointer to an Impl. Then this impl is created by a function dependant on the platform ? Almost what's here except with another level of abstraction. I don't see the point here.

Reply:
nit: It should be the same amount of indirection, just without a static factory and with pImpl instead of virtuality. So the `CameraInterface` declaration becomes
```
class CameraInterface
{
public:
    CameraInterface(Napi::Env, uint32_t, uint32_t, bool);
    void UpdateCameraTexture(bgfx::textureHandle);

private:
    struct Impl;
    std::unique_ptr<Impl> m_impl{};
}
```
and the definitions of everything, most importantly `Impl`, are platform-specific. At runtime, this should be basically the same as what you already have, so I don't think this is a huge deal either way. This approach just sticks closer to the patterns we've used most commonly elsewhere (avoiding both virtuality and staticness) and may help clarify from a contract standpoint that there's never actually more than one kind of `CameraInterface` in any given build. So, while I do think pImpl is better here, I'm not really concerned about it; the raw pointer was the only thing that originally made this comment more than a nit.

Reply:
I would echo @syntheticmagus's comments here, but probably a bit more strongly. I think the inheritance pattern here is actually misleading, because it suggests there is a *runtime* abstraction, when in fact there is only a *compile* time abstraction. I think it would be much easier to quickly understand the code if it followed the same patterns we use elsewhere, both for the directory structure for platform specific vs. shared files and the pimpl pattern.